graph_weather/data/weather_station_reader.py

```
@@ -0,0 +1,772 @@
+                    elevation = station_info.get("ELEVATION")
+
+                    # Store metadata
+                    if "lat" not in data_vars:
+                        data_vars["lat"] = (["station"], [lat for _ in stations])
+                    if "lon" not in data_vars:
+                        data_vars["lon"] = (["station"], [lon for _ in stations])
+                    if "elevation" not in data_vars:
+                        data_vars["elevation"] = (["station"], [elevation for _ in stations])
+
+                    # Process observation variables
+                    if "OBSERVATIONS" in station_info:
+                        obs = station_info["OBSERVATIONS"]
+
+                        # For each variable except date_time
+                        for var_name, values in obs.items():
+                            if var_name != "date_time":
+                                # Initialize if needed
+                                if var_name not in data_vars:
+                                    var_array = np.full((len(times), len(stations)), np.nan)
+                                    data_vars[var_name] = (["time", "station"], var_array)
+
+                                # Fill data
+                                for i, dt in enumerate(obs["date_time"]):
+                                    if i < len(values):  # Check bounds
+                                        time_idx = times.index(dt)
+                                        station_idx = stations.index(station_id)
+                                        data_vars[var_name][1][time_idx, station_idx] = values[i]
+
+                # Create the dataset
+                ds = xr.Dataset(data_vars=data_vars, coords=coords)
+                return ds
+            else:
+                logger.error("Unexpected response format from SynopticPy")
+                return None
+        except Exception as e:
+            logger.error(f"Error fetching data from SynopticPy: {str(e)}")
```

I would raise the error here, not return None, as if we call fetch synoptic and we don't get data, we want the error to propogate.